### PR TITLE
[WIP] New docs

### LINKED ISSUE #3: Switch from Redcarpet to Kramdown
> **Starting May 1st, 2016, GitHub Pages will only support kramdown, Jekyll's default Markdown engine.**

From https://github.com/blog/2100-github-pages-now-faster-and-simpler-with-jekyll-3-0.

### LINKED ISSUE #24: Add documentation on how to proxy behind common web servers
http://nginx.org/en/docs/http/websocket.html
https://stackoverflow.com/questions/27526281/websockets-and-apache-proxy-how-to-configure-mod-proxy-wstunnel

### LINKED ISSUE #42: Include HTTPS setup in the docs
Someone wrote a documentation about setting up HTTPS for The Lounge with self-signed certificates: https://nertant.com/2016/03/12/enabling-https-ssl-connection-with-the-lounge-self-hosted-irc-client/

Similarly, @YaManicKill made a wiki page some time ago about setting up Let's Encrypt certificates: https://github.com/thelounge/lounge/wiki/Enable-HTTPS-with-Let's-Encrypt

We should probably document this properly and "officially" in this repository instead.

We might want to answer the following questions first:
- Do we want to document about self-signed certificates? Actual CA-approved certificates? Both?
- Do we use Let's Encrypt as an example (therefore, implicitly recommending it) or assume reader comes with their own certificate?
- Where do we locate this in the docs?

### LINKED ISSUE #56: Add "Edit this page" on each documentation page

### LINKED ISSUE #57: Document how to create a standalone window in Chrome

### LINKED ISSUE #71: Remove documentation of --home and give more details about LOUNGE_HOME
This is a reminder for myself to add this to the [new documentation](https://github.com/thelounge/thelounge.github.io/tree/astorije/better-docs).

In the meantime, I'll mark it as deprecated on the current website, like we did for #61.

### LINKED ISSUE #83: Switch lounge to thelounge in documentation
`lounge` is deprecated in CLI, it is recommended to use `thelounge`. Documentation should be changed to reflect that.

### PR DESCRIPTION
**Pull request by astorije**
This is still a work in progress, but keeping it as an open PR as we had people wonder why the docs don't reflect the current state of the project. This will also generate a preview for whoever wants to access them (at your own risk :D).

Definitely not ready for review, so don't waste your time on it yet :)

~Closes #42.~ _Extracted to #42_
~Closes #56.~ _Extracted to #134_
~Closes #57.~ _Nope_
~Closes #3.~ _Extracted to #91_
~Closes #24.~ _Extracted to #119_
~Closes #71.~ _Extracted to #116_.
~Closes #83.~ _Extracted to #104_

### TODO

- [x] Update favicon #98
- [x] Move search index to its own file instead of embedding in html #101 
- [x] "IRC" has two underlines in docs (one from link, one from abbr)
- [x] Guide names shouldn't all start with "How to...", redundant and constraining.
- [x] Pre-release install button takes full-width on homepage #137 
- Mobile / Responsiveness issues:
  - ~Community page: Backers snippet goes out of the screen~ #138 
  - ~Community page: Youtube embed goes out of the page~ #138 
  - ~Configuration page: Long `ldapFilter` string is not breaking to the next line~ #151 
  - [x] Header and menu look poor
  - [x] "Edit this page" button weirdly aligns on most pages #56 
- Update the upgrade guide: #124 
  - [x] Mention logs timestamps are now UTC (see [this](https://github.com/thelounge/thelounge/pull/2787/files#r219361434)). Check if they weren't UTC before v3 anyway.
  - [x] Docker breaking changes
  - [x] Node.js v9 is not supported
- ~Menu does not mark Guides/API as `active` when visiting one of the guides or the APIs~ _Moved to https://github.com/thelounge/thelounge.github.io/issues/156_

### Won't fix

- Color contrast on links are not AA compliant: Will do on re-style post-docs

### CODE REVIEW

**Comment by McInkay** [comment]
An additional thing we need to add to the todo list: theme/package instructions. You've got the API but not instructions on installing/using.

**Comment by astorije** [comment]
@McInkay, thanks for checking out! :)
I did add something actually, in the Usage section: https://deploy-preview-84--thelounge-chat.netlify.com/docs/usage#installing-additional-themes

At the moment, it's only about themes because there are no packages out there at all, but as soon as https://www.npmjs.com/search?q=keywords%3Athelounge-package has a single package, I'll update the section to be about both themes and packages :)
Chicken and egg situation, because it could also be confusing to tell users how to install packages when there is not a single one.
I am confident you will be publishing the very first package, so I'm happy to work on the doc if you do so!

**Comment by McInkay** [comment]
I have a number of packages ready to be published once my PRs are merged :-P so yeah, that's fine leaving it as just themes just now, I get that.

However, I'm wondering whether that's obvious enough or not. Maybe it is.

**Comment by xPaw** [comment]
It should be noted in install docs that since Yarn installs into user folder by default, the binary isn't going to be available in $PATH.

On linux the fix is to add Yarn global path to your $PATH like `export PATH="$PATH:$(yarn global bin)"`

**Comment by astorije** [comment]
@xPaw, I'd rather not do this because it's very platform-dependent. For example, on macOS (though I know server install is rarely on macOS, but still, we do advertise it as cross-OS :D), it's available by default if you install with Homebrew.

You probably remember, but long time ago ([ahem](http://shout-irc.com/docs/getting_started/install.html)) we were documenting how to install Node on Ubuntu, Mac, and Windows, then we had to add CentOS/Fedora/RHEL, at which point we decided to redirect to [Node's comprehensive package manager install document](https://nodejs.org/en/download/package-manager/) (as it is on the [current doc](https://thelounge.chat/docs/getting_started/install.html)). The [Yarn install page is actually really good](https://yarnpkg.com/lang/en/docs/install/), lets you pick your OS and gives instructions accordingly, so I think I'd rather send reader to this page early in the install section, similarly to what we do with Node. You'd be okay with that?

**Comment by astorije** [comment]
Yep, that's what I meant, adding a link to this page and making it all a bit better. Will do!

File: _api/stylesheets.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+---
+layout: documentation
+title: Stylesheets
+description: Subscribe CSS files to to all clients
```
[the remark this answers is not in the document]
Comment (astorije) [reply _api/stylesheets.md]: Thanks, fixed the typo. And I agree with you re: "subscribe", I remember struggling a bit with this one. I'm not sure about "include", but happy to go with that. Soon enough I'll extract this page/part of the doc to its own PR, we can fix/improve content then.
Thanks! 🙏

File: _docs/install-and-upgrade.md [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+Its configuration file is located at `~/.thelounge/config.js`. To configure The
+Lounge, go to [the configuration section](/docs/configuration).
+
+To upgrade The Lounge, simply re-run the `install` command above, and restart it.
+
+You might want to serve The Lounge behind a reverse proxy such as Nginx. To know
+more about the benefits and steps, follow
+[the guide on reverse proxies](/docs/guides/reverse-proxies).
+
+## Docker
```
Comment (williamboman) [inline _docs/install-and-upgrade.md]: I'm sure you're aware of this but flagging anyway - this section is outdated now

Comment (xPaw) [reply _docs/install-and-upgrade.md]: Yeah this PR was split up into multiples ones, so when this is rebased, this change will be gone.

Updated documentation is already on master.

File: Gemfile [hunk 3]
```diff
@@ -1,4 +1,5 @@
 source "https://rubygems.org"
 
 gem 'jekyll', '3.8.4'
+gem 'jekyll-redirect-from', '0.14.0'
```
Comment (xPaw) [inline Gemfile]: How about we use `_redirects` file instead of bringing in this gem?

It's also much nicer to see all redirects in a single place.

Comment (astorije) [reply Gemfile]: Good point, switching to that

File: _guides/reverse-proxies.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+	proxy_set_header X-Forwarded-For $remote_addr;
+
+	# by default nginx times out connections in one minute
+	proxy_read_timeout 1d;
+}
+```
+
+## [Apache](https://httpd.apache.org/)
+
+Enable the necessary modules `a2enmod rewrite`, `a2enmod proxy`, and
```
Comment (xPaw) [inline _guides/reverse-proxies.md]: Apparently `a2enmod proxy_http` is also needed.

File: css/style.css [hunk 5]
```diff
@@ -430,16 +440,113 @@ a.link-anchor {
...
+	font-size: .9em;
+}
+
+/* Terminal windows-style, to present CLI outputs and such */
+
+.terminal {
+	background-color: black;
+	border-radius: 5px;
+	color: #c7c7c7;
+	font-family: monospace;
```
Comment (xPaw) [inline css/style.css]: Add `.terminal` selector where I recently added monospace fonts

File: _config.yml [hunk 6]
```diff
@@ -6,14 +6,17 @@ include:
   - _redirects
 kramdown:
   toc_levels: "2"
+include:
+  - _redirects
```
Comment (xPaw) [inline _config.yml]: dupe

**Comment by astorije** [comment]
Closing this PR now. Everything has been moved to other PRs/issues and in most cases merged. Only leftovers of this PR are things that do not need to be merged.

**Pull request closed without merging**